src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php

```diff
@@ -1023,4 +1024,74 @@ private function createUninitializedObjectPropertyGhost(): UninitializedObjectPr
         return $class::createLazyGhost(initializer: function ($instance) {
         });
     }
+
+    /**
+     * @requires PHP 8.4
+     */
+    public function testIsWritableWithAsymmetricVisibility()
+    {
+        $object = new AsymmetricVisibility();
+
+        $this->assertTrue($this->propertyAccessor->isWritable($object, 'publicPublic'));
+        $this->assertFalse($this->propertyAccessor->isWritable($object, 'publicProtected'));
+        $this->assertFalse($this->propertyAccessor->isWritable($object, 'publicPrivate'));
+        $this->assertFalse($this->propertyAccessor->isWritable($object, 'privatePrivate'));
+        $this->assertFalse($this->propertyAccessor->isWritable($object, 'virtualNoSetHook'));
+    }
+
+    /**
+     * @requires PHP 8.4
+     */
+    public function testIsReadableWithAsymmetricVisibility()
+    {
+        $object = new AsymmetricVisibility();
+
+        $this->assertTrue($this->propertyAccessor->isReadable($object, 'publicPublic'));
+        $this->assertTrue($this->propertyAccessor->isReadable($object, 'publicProtected'));
+        $this->assertTrue($this->propertyAccessor->isReadable($object, 'publicPrivate'));
+        $this->assertFalse($this->propertyAccessor->isReadable($object, 'privatePrivate'));
+        $this->assertTrue($this->propertyAccessor->isReadable($object, 'virtualNoSetHook'));
+    }
+
+    /**
+     * @requires PHP 8.4
+     */
+    public function testSetValueWithAsymmetricVisibility()
+    {
+        $object = new AsymmetricVisibility();
+
+        $this->propertyAccessor->setValue($object, 'publicPublic', true);
+        $this->assertTrue($object->publicPublic);
+
+        try {
+            $this->propertyAccessor->setValue($object, 'publicProtected', true);
+            $this->fail('Property with protected set should not be writable.');
```

Comment:
Instead of a single test performing such try/catch with explicit failures if there is no exception, please use the `expectException` API of PHPUnit, which provides a much better experience when dealing with test failure.
This will require separating each tested case in a separate test instead of a single test covering all cases. But this is actually a good thing as it will make the intent of tests easier to understand as well.